packages/components/src/components/Snackbar/useSnackbar.tsx

```diff
@@ -15,23 +16,36 @@ export function useSnackbar() {
   const toast = Toast.useToast()
   const context = useContext(SnackbarContext)
   const defaultOffset = useSnackbarDefaultOffset()
+  const screenReaderEnabled = useIsScreenReaderEnabled()
 
   if (!context) {
     throw new Error('useSnackbar must be used within a SnackbarProvider')
   }
 
-  const { offset, setOffset } = context
-
   const show = (message: string, snackbarOptions?: SnackbarOptions) => {
-    toast.hideAll()
+    const { offset, setOffset, duration, setDuration } = context
 
-    if (snackbarOptions?.offset) {
-      setOffset(snackbarOptions.offset)
-    } else if (offset !== defaultOffset) {
-      setOffset(defaultOffset)
+    // Adjust offset if provided or reset to default
+    const newOffset = snackbarOptions?.offset || defaultOffset
+
+    // Only call setOffset if different from current to avoid re-render
```

Comment:
Is this a problem if `toast.hideAll()` was still at the top of the function? Or is the re-render with the SnackbarProvider, not the Snackbar itself?

Reply:
No problem or functional difference. Just moved it while cleaning up the funciton.